mirrord-layer/src/config.rs

```diff
@@ -67,3 +71,105 @@ pub struct LayerConfig {
     #[envconfig(from = "MIRRORD_SKIP_PROCESSES")]
     pub skip_processes: Option<String>,
 }
+
+#[derive(Debug, PartialEq)]
+pub struct Deployment(pub String);
+
+#[derive(Debug, PartialEq)]
+pub enum Target {
+    Deployment(Deployment),
+    Pod(PodAndContainer),
+}
+
+impl Target {
+    pub async fn container_info(
+        &self,
+        pods_api: &Api<Pod>,
+        client: &Client,
+        pod_namespace: &str,
+    ) -> Option<String> {
+        match self {
+            Target::Pod(PodAndContainer(pod, container)) => {
+                self.pod(&pods_api, pod, container, pod_namespace, &client)
+                    .await
+            }
+            Target::Deployment(_) => None,
+        }
+    }
+
+    pub async fn pod(
+        &self,
+        pods_api: &Api<Pod>,
+        pod_name: &String,
+        container_name: &Option<String>,
+        pod_namespace: &str,
+        client: &Client,
+    ) -> Option<String> {
+        let pod = pods_api.get(pod_name).await.unwrap();
+        let node_name = &pod.spec.unwrap().node_name;
+        let container_statuses = &pod.status.unwrap().container_statuses.unwrap();
+        let container_info = if let Some(container_name) = container_name {
+            &container_statuses
+                .iter()
+                .find(|&status| &status.name == container_name)
+                .ok_or_else(|| {
+                    LayerError::ContainerNotFound(
+                        container_name.clone(),
+                        pod_namespace.to_string(),
+                        pod_name.to_string(),
+                    )
+                })
+                .unwrap()
+                .container_id
+        } else {
+            info!("No container name specified, defaulting to first container found");
+            &container_statuses.first().unwrap().container_id
+        };
+        container_info.clone()
+    }
+
+    pub fn deployment(&self) -> Option<String> {
+        todo!()
+    }
+
+    pub async fn node_name(&self, pods_api: Api<Pod>) -> Option<String> {
+        match self {
+            Target::Pod(PodAndContainer(pod_name, _)) => {
+                let pod = pods_api.get(&pod_name).await.unwrap();
+                pod.spec.unwrap().node_name
+            }
+            Target::Deployment(_) => None,
+        }
+    }
+}
+
+impl FromStr for Deployment {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let split = s.split('/').collect::<Vec<&str>>();
+        match split.first() {
+            Some(&"deployment") => Ok(Deployment(split[1].to_string())),
+            _ => Err(format!("Given target: {:?} is not a deployment.", s)),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct PodAndContainer(pub String, pub Option<String>);
+
+impl FromStr for PodAndContainer {
+    type Err = String;
```

Comment:
`Err` should be a proper `Error`, not `String`, for example (feel free to use a better name).

```suggestion
    type Err = ParsePodNameError;
```